app/lib/provider_discovery.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ProviderDiscovery < OEmbed::ProviderDiscovery
+  USER_AGENT = "#{HTTP::Request::USER_AGENT} (Mastodon/#{Mastodon::VERSION}; +http://#{Rails.configuration.x.local_domain}/)"
+
+  class << self
+    def discover_provider(url, options = {})
+      res    = http_client.get(url)
+      format = options[:format]
+
+      raise OEmbed::NotFound, url if res.code != 200 || res.mime_type != 'text/html'
+
+      html = Nokogiri::HTML(res.to_s)
+
+      if format.nil? || format == :json
+        provider_endpoint = html.at_xpath('//link[@type="application/json+oembed"]')['href']
+        format ||= :json
+      end
+
+      if format.nil? || format == :xml
```

Comment:
The first conditional here will never get called, since if format is nil, we set it to :json on line 17.